src/sign_msg.c

```diff
@@ -150,11 +150,10 @@ void handle_sign_msg(uint8_t p1,
 
     // finalize hash, compute it and store it in `msg_context.strhash` for display
     cx_hash((cx_hash_t *) &sha3_context, CX_LAST, data_buffer, 0, msg_context.hash, HASH_LEN);
-    snprintf(msg_context.strhash,
-             sizeof(msg_context.strhash),
-             "%.*H",
-             sizeof(msg_context.hash),
-             msg_context.hash);
+    convert_to_hex_str(msg_context.strhash,
```

Comment:
I think that this code here is not covered by the existing tests so we don't have a way currently to see if the hash display is broken

Reply:
The code is covered by `test_sign_msg_long` and `test_sign_msg_short`. (they failed a few times during the refactoring until they passed again)